flocker/control/_config.py

```diff
@@ -1283,7 +1295,7 @@ def model_from_configuration(applications, deployment_configuration):
     :returns: A ``Deployment`` object.
     """
     nodes = deployment_from_configuration(
-        deployment_configuration, applications)
+        deployment_state, deployment_configuration, applications)
     return Deployment(nodes=frozenset(nodes))
```

Comment:
Yeah, but I think it's time to switch to "do things as quickly as possible" and do cleanup later.